certego_saas/ext/upload/twitter.py

```diff
@@ -30,69 +30,68 @@ def post_tweet(
-
-    def post_tweet(
-        self,
-        msg: str,
-        media: List[Union[str, models.FileField]] = None,
-        header: str = None,
-    ):
+    class Twitter(_TwitterInterface):
         """
-        To post a tweet.
+        Twitter client.
         """
-        urls = self.__parse(msg)
-        msg = strip_tags(msg)
-        msg += " ".join(urls)
-        # i wanna add a custom delimiter + initial message
-        if header:
-            msg = header + " " + msg
-        size = self.CHARACTER_LIMIT - 9
-        if len(msg) >= self.CHARACTER_LIMIT:
-            # splitting the text in N messages
-            messages = [msg[i : i + size] for i in range(0, len(msg), size)]
-            # adding 0/10, 1/10 to the end of the messages
-            messages = [
-                msg + f" {i}/{len(messages) - 1}" for i, msg in enumerate(messages)
-            ]
-        else:
-            messages = [msg]
-        self.log.info(messages)
-        result_id = self.client.PostUpdate(status=messages[0], media=media).id
-        for msg in messages[1:]:
-            result_id = self.client.PostUpdate(
-                status=msg, in_reply_to_status_id=result_id
-            ).id
-
 
-#: Twitter Client
-Twitter = _FakeTwitter if settings.STAGE_LOCAL or settings.STAGE_CI else _Twitter
+        CHARACTER_LIMIT = twitter_lib.api.CHARACTER_LIMIT
+        client = twitter_lib.Api(
+            consumer_key=certego_apps_settings.TWITTER_CONSUMER_KEY,
+            consumer_secret=certego_apps_settings.TWITTER_CONSUMER_SECRET,
+            access_token_key=certego_apps_settings.TWITTER_TOKEN_KEY,
+            access_token_secret=certego_apps_settings.TWITTER_TOKEN_SECRET,
+        )
+
+        def __init__(self):
+            if not self.client.VerifyCredentials():
+                raise Exception("Wrong credentials")
+
+        @staticmethod
+        def __parse(msg) -> List[str]:
+            return re.findall(r'href=[\'"]?([^\'" >]+)', msg)
+
+        def post_tweet(
+            self,
+            msg: str,
+            media: List[Union[str, models.FileField]] = None,
+            header: str = None,
+        ):
+            """
+            To post a tweet.
+            """
+            urls = self.__parse(msg)
+            msg = strip_tags(msg)
+            msg += " ".join(urls)
+            # i wanna add a custom delimiter + initial message
+            if header:
+                msg = header + " " + msg
+            size = self.CHARACTER_LIMIT - 9
+            if len(msg) >= self.CHARACTER_LIMIT:
+                # splitting the text in N messages
+                messages = [msg[i : i + size] for i in range(0, len(msg), size)]
+                # adding 0/10, 1/10 to the end of the messages
+                messages = [
+                    msg + f" {i}/{len(messages) - 1}" for i, msg in enumerate(messages)
+                ]
+            else:
+                messages = [msg]
+            self.log.info(messages)
```

Comment:
this log is not searchable without additional text. It was an already existing problem. And no other logs are here too